### PR TITLE
Add workflow recommendations and templates

### PR DESCRIPTION
**Pull request by SleeplessByte**
Based on all the feedback I got from @SaschaMann, I've created a baseline set of workflows anyone can use. This is meant to set-up the correct triggers and files, NOT the actual CI (how to run the test/how to do x y z).

1. I'm not looking for a long debate on how to optimise these further. This is meant to be an easy way for maintainers to get started with GitHub Actions, without having to do the research themselves. I invite further optimisations to be PR-ed in.
2. This is not a _great_ place for these files, but it's a good place for discoverability. Recommended by @iHiD , so we're good on that.
3. I'm perfectly fine with changing wording, but please make a suggestion so it can be applied quickly, and then we can merge it in and make further improvements in future PRs.
4. I'm not looking for a debate on where the scripts/binaries should be placed. Common places for scripts is `scripts`, common places for binaries is `bin`. Since it's _literally_ only changing 1 line per script in the workflow, it's not worth debating. I purposefully made the migration example use `bin/`, and the other example use `scripts/` to further establish that both are fine.

In short, the readme should be explanatory; you can drop-in the workflow files into your track, replace the template variables, and add the four scripts / bin files (**you need to write these yourselves, in your language of choice**) and it "just works".

### CODE REVIEW

**Review by SaschaMann — COMMENTED** [review]
This is a great resource :)

Org-wide workflows are on the GH roadmap, hopefully once that feature is we could centralise things like the configlet checks.

File: workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  workflow_dispatch:
+
+jobs:
+  precheck:
+    runs-on: <image-name>
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use <setup tooling>
+        uses: <action to setup tooling>
```
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply workflows/ci.yml:27]: Good idea! Maybe just as URLs inside the comment block at top?

File: workflows/README.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+## Recommendations
+
+For each action think about how often it should run.
+
+- The `configlet` linting is something that's so important (because a track can break if the `config.json` breaks), that it probably should run, always, **but** only needs to run once per commit.
+- The existence or integrity of files only needs to run once per commit.
+- If a track is supposed to run under multiple runtime-versions or compiler-versions, building/testing exercises should be ran against each supported version
+- PRs _probably_ only need to run actions on files _added_ or _changed_, but since a file can influence an exercise, it's safer to run the actions for the _exercise_, if one of its files changes.
+
+It can be very helpful to make the actions that should run, available locally as well. This means that the scripts that do the actual work are also manually runnable. **Do not inline** the action inside the workflow files. For example, checking for stubs can be completely bashed out inside the workflow file, but the recommendation here is to create a new executable script `scripts/ci-check` instead.
```
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply workflows/README.md:38]: It's necessary to allow people to run _exactly_ the same command. 

> In some exercism repos those scripts can also be found in bin/ or .github/bin.

Yep, see further down.

File: workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+          # node-version: 12.x
+
+      - name: Install project dependencies
+        run: <install dependencies>
+
+      - name: Run exercism/<track> ci pre-check (checks config, lint code) for all exercises
+        run: scripts/ci-check
+
+  ci:
+    runs-on: <image-name>
```
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply workflows/ci.yml:39]: Good idea!

File: workflows/README.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+If the track has a single "top-level" dependency file and/or other configuration files, add an integrity step (to exist alongside a `scripts/sync` or `bin/sync`) that ensures the top-level/base files are the same as the one copied to the exercise directories. Now dependencies can be updates, synced across the repository, and ensures that all exercises have the same configuration.
+
```
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply workflows/README.md:41]: I like this. Thanks.

File: workflows/README.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+```bash
+git update-index --chmod=+x scripts/xxx
+git commit -m "Make scripts/xxx executable"
+```
+
+[git-configlet]: https://github.com/exercism/configlet#lint
+[git-index]: https://www.git-scm.com/docs/git-update-index
+[git-javascript]: https://github.com/exercism/javascript/tree/f49ac022d3a55cbbb48dadbc6dbf1d407de72187/.github/workflows
+[github-workflow-dispatch]: https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/
```
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply workflows/README.md:81]: Great resource!

File: README.md [hunk 6]
```diff
@@ -51,83 +51,90 @@ Test data can be incorporated into a track's test suites manually or extracted b
...
-        }
-      , "expected"   : "noil"
-      }
-    , { "uuid"       : "09113ce5-b008-45d0-98af-c0378b64966b"
-      , "description": "Bar'ing a name returns its parts combined"
-      , "property"   : "bar"
-      , "input"      : {
-          "firstName"  : "Alan",
-          "lastName"   : "Smithee"
+{
```
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply README.md:54]: Haha yeah, fucking autoformat 🗡️

File: workflows/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply workflows/README.md:38]:
```suggestion
It can be very helpful to make the actions that should run, available locally as well. This means that the scripts that do the actual work are also manually runnable. **Do not inline** the action inside the workflow files. For example, checking for stubs can be completely bashed out inside the workflow file, but the recommendation here is to create a new executable script `scripts/ci-check` instead.

> "But the command is very short, e.g. `eslint . --ext ts --ext tsx`".
>
> When this command needs to be updated, it now needs to update in all the places in the documentation, the workflow files, ánd in the _minds of the maintainers_. Extracting this to a script resolves all that. Reading a workflow file can also be **very** daunting.
```

File: workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (SaschaMann) [reply workflows/ci.yml:27]: That works, I wasn't sure where to put it

File: workflows/README.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+Each file has listed at the top which "scripts" should be available. If you want these to be binaries, replace `scripts/xxx` with `bin/xxx`. Some tooling will _require_ binaries to be inside a `bin` folder.
+
+- `scripts/ci`: a script that should build and test all exercises using the example solutions against the tests
+- `scripts/ci-check`: a script that should lint all exercises, and optionally check for stubs, configuration integrity, and more
+- `scripts/pr`: same as `scripts/ci`, but should only run exercises resolved from the paths given as input
+- `scripts/pr-check`: same as `scripts/ci-check`, but should only run for files or exercises resolved from the paths given as input
+
+## Troubleshooting
+
```
Comment (SaschaMann) [inline workflows/README.md:64]:
```suggestion
## Troubleshooting

If you run into any issues or want someone to review your workflows, please ping the `@exercism/github-actions` team.

```

File: workflows/ci.yml [hunk 8]
```diff
@@ -0,0 +1,58 @@
+# This workflow will do a clean install of node dependencies and run tests across different versions
+#
+# Replace <track> with the track name
+# Replace <image-name> with an image to run the jobs on
+# Replace <action to setup tooling> with a github action to setup tooling on the image
+# Replace <install dependencies> with a cli command to install the dependencies
+# Replace <code-extensions> with file extensions that should trigger the workflow
+#
```
Comment (SleeplessByte) [inline workflows/ci.yml:8]:
```suggestion
#
# Find Github Actions to setup tooling here:
# - https://github.com/actions/?q=setup&type=&language=
# - https://github.com/actions/starter-workflows/tree/main/ci
# - https://github.com/marketplace?type=actions&query=setup
#
```

File: workflows/ci.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply workflows/ci.yml:39]: Hmm. It does auto label right now for me. Is this different other than adding dashes? @SaschaMann ?

File: workflows/README.md [hunk 9]
```diff
@@ -4,6 +4,10 @@ The content in this folder doesn't _necessarily_ belong in this repository, but
 
 Example implementation of these workflow files [can be found in `exercism/javascript`][git-javascript].
 
+## **HELP** this looks like a lot of work
+
+The rest of the document is really to explain what's going on in the workflows. If you're in a hurry, and you really want to drop `.travis.yml`, but you can't invest right now in making the PR scripts are optimized, scroll down to **Migrating from Travis**.
```
Comment (SaschaMann) [inline workflows/README.md:9]:
```suggestion
The rest of the document is really to explain what's going on in the workflows. If you're in a hurry, and you really want to drop `.travis.yml`, but you can't invest right now in making the PR scripts are optimized, scroll down to [**Migrating from Travis**](#migrating-from-travis).
```

(not 100% sure this is the right syntax)

Comment (SleeplessByte) [reply workflows/README.md:9]: (not sure either, but we can try)

File: workflows/ci.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (SaschaMann) [reply workflows/ci.yml:44]: Not necessary but imo it looks nicer to explicitely set a name

[the remark this answers is not in the document]
Comment (SleeplessByte) [reply workflows/ci.yml:44]: Cool. I'll add instructions on how to do this.

**Review by neenjaw — APPROVED** [review]
Nice work, this looks thorough. Now to implement it for elixir....

**Review by SaschaMann — APPROVED** [review]

File: workflows/README.md [hunk 10]
```diff
@@ -0,0 +1,310 @@
+# Workflow templates
+
+The content in this folder doesn't _necessarily_ belong in this repository, but we don't have a great place to put it that allows for good discoverability. The GitHub Actions workflows in this folder can be adapted to work with any CI, because the base structure will remain the same.
```
Comment (iHiD) [inline workflows/README.md:3]: Suggest having an overarching comment about why someone should even read this, and to establish a couple of acronyms we can reuse.

```suggestion
This document explains how to set up Continuous Integration (CI) workflows for an Exercism Language Track using GitHub Actions (GHA). It provides best practices and examples for you to use to make your own fast, reliable and robust CI workflows. The GHA workflows in this folder can be adapted to work with any CI, because the base structure will remain the same.

_Note: The content in this folder doesn't **necessarily** belong in this repository, but we don't have a great place to put it that allows for good discoverability. _
```

File: workflows/README.md [hunk 11]
```diff
@@ -0,0 +1,310 @@
...
+
+Example implementation of these workflow files [can be found in `exercism/javascript`][git-javascript].
+
+## **HELP**: this looks like _a lot_ of work :sweat:
+
+The rest of the document is really to explain what's going on in the workflows. If you're in a hurry, and you really want to drop `.travis.yml`, but you can't invest right now in making the PR scripts are optimized, scroll down to the **~10 minute guide** on [**Migrating from Travis**](#migrating-from-travis).
```
Comment (iHiD) [inline workflows/README.md:9]:
```suggestion
The rest of the document is designed to explain how the workflows work. If you're in a hurry, and you just want to switch from Travis or Circle to GHA without optimizing the PR scripts, scroll down to the **~10 minute guide** on [**Migrating from Travis**](#migrating-from-travis).
```

File: workflows/README.md [hunk 12]
```diff
@@ -0,0 +1,310 @@
...
+
+## Track CI actions
+
+The recommended actions are as follows:
```
Comment (iHiD) [inline workflows/README.md:13]:
```suggestion
The recommended actions for checking the content of your repository has integrity are as follows:
```

File: workflows/README.md [hunk 13]
```diff
@@ -0,0 +1,310 @@
...
+
+1. [`configlet` linting][git-configlet] in order to check `config.json`
+2. check for stubs
+3. check for documentation (`v3` requires new files; this might move to `configlet`)
+4. lint the exercises using a "maintainers" configuration
+5. test the exercises using the example/examplar files (can include build step)
+
+There can be track-specific actions, for example:
```
Comment (iHiD) [inline workflows/README.md:21]:
```suggestion
There can also be track-specific actions. For example:
```

File: workflows/README.md [hunk 14]
```diff
@@ -0,0 +1,310 @@
...
+1. ensure CONTRIBUTING exists
+2. ensure there is a sane lockfile for dependencies
+3. ensure links inside markdown files are valid
+4. ...
+
+## Recommendations
+
+For each action think about how often it should run.
+
+- The `configlet` linting is something that's so important (because a track can break if the `config.json` breaks), that it probably should run, always, **but** only needs to run once per commit.
```
Comment (iHiD) [inline workflows/README.md:37]:
```suggestion
- The `configlet` linting is something that's so important (because a track can break if the `config.json` breaks) that it should probably always run, **but** only needs to run once per commit.
```

File: workflows/README.md [hunk 15]
```diff
@@ -0,0 +1,310 @@
...
+- The existence or integrity of files only needs to run once per commit.
+- If a track is supposed to run under multiple runtime-versions or compiler-versions, building/testing exercises should be ran against each supported version
+- PRs _probably_ only need to run actions on files _added_ or _changed_, but since a file can influence an exercise, it's safer to run the actions for the _exercise_, if one of its files changes.
+
+It can be very helpful to make the actions that should run, available locally as well. This means that the scripts that do the actual work are also manually runnable. **Do not inline** the action inside the workflow files. For example, checking for stubs can be completely bashed out inside the workflow file, but the recommendation here is to create a new executable script `scripts/ci-check` instead.
```
Comment (iHiD) [inline workflows/README.md:42]:
```suggestion
It can be very helpful to make the actions that should run, available locally as well. This means that the scripts that do the actual work are also manually runnable. To achieve this **do not inline** the action inside the workflow files., but create a standalone script. For example, checking for stubs can be completely bashed out inside the workflow file, but the recommendation here is to create a new executable script `scripts/ci-check` instead.
```

File: workflows/README.md [hunk 16]
```diff
@@ -0,0 +1,310 @@
...
+
+> "But the command is very short, e.g. `eslint . --ext ts --ext tsx`".
+>
+> When this command needs to be updated, it now needs to update in all the places in the documentation, the workflow files, ánd in the _minds of the maintainers_. Extracting this to a script resolves all that. Reading a workflow file can also be **very** daunting.
```
Comment (iHiD) [inline workflows/README.md:46]: Not sure if this is intended to all be blockquoted or not (both work)

```suggestion
When this command needs to be updated, it now needs to update in all the places in the documentation, the workflow files, ánd in the _minds of the maintainers_. Extracting this to a script resolves all that. Reading a workflow file can also be **very** daunting.
```

File: workflows/README.md [hunk 17]
```diff
@@ -0,0 +1,310 @@
...
+
+### PR exercises changed
+
+The `scripts/pr` and `scripts/pr-check` scripts are ran with arguments: each file that has been changed or added in this PR. For example, if `two-fer` has been updated, a call might look like this:
```
Comment (iHiD) [inline workflows/README.md:50]:
```suggestion
The `scripts/pr` and `scripts/pr-check` scripts are run with `n` arguments, one for each file that has been changed or added in this PR. For example, if `two-fer` has been updated, a call might look like this:
```

File: workflows/README.md [hunk 18]
```diff
@@ -0,0 +1,310 @@
...
+
+```bash
+scripts/pr exercises/two-fer/README.md exercises/two-fer/.meta/example.ext
+```
+
+It's recommended to run any actions against the changed _exercise_ and not the changed _file_. This is because changing a file is likely to trigger changes for the entire exercise (think: configuration, packages).
+
+> **Not ready?** / **Complex?**
+>
+> Before implementing this optimization, it may be ignored! The migration guide hints at adding at a later stage. If the input arguments are ignored, all the checks are running on all the exercises. **This is perfectly fine**. It will just take longer.
```
Comment (iHiD) [inline workflows/README.md:60]:
```suggestion
> Before implementing this optimization, it may be safely ignored! The migration guide hints at adding at a later stage. If the input arguments are ignored, all the checks will run on all the exercises. **This is perfectly fine**. It will just take longer.
```

File: workflows/README.md [hunk 19]
```diff
@@ -0,0 +1,310 @@
...
+The rest of the document is really to explain what's going on in the workflows. If you're in a hurry, and you really want to drop `.travis.yml`, but you can't invest right now in making the PR scripts are optimized, scroll down to the **~10 minute guide** on [**Migrating from Travis**](#migrating-from-travis).
+
+## Track CI actions
+
+The recommended actions are as follows:
+
+1. [`configlet` linting][git-configlet] in order to check `config.json`
+2. check for stubs
+3. check for documentation (`v3` requires new files; this might move to `configlet`)
+4. lint the exercises using a "maintainers" configuration
```
Comment (iHiD) [inline workflows/README.md:18]: Should this be "maintainers'" (ie its the configuration the maintainers have made). I wonder if it would be better renamed to "track-specific configuration"?

File: workflows/README.md [hunk 20]
```diff
@@ -0,0 +1,310 @@
...
+And perhaps you'd want more Quality Of Life checks, such as:
+
+1. ensure CONTRIBUTING exists
+2. ensure there is a sane lockfile for dependencies
+3. ensure links inside markdown files are valid
+4. ...
+
+## Recommendations
+
+For each action think about how often it should run.
```
Comment (iHiD) [inline workflows/README.md:35]:
```suggestion
### Frequency of running checks

For each action think about how often it should run.
```

File: workflows/README.md [hunk 21]
```diff
@@ -0,0 +1,310 @@
...
+- If a track is supposed to run under multiple runtime-versions or compiler-versions, building/testing exercises should be ran against each supported version
+- PRs _probably_ only need to run actions on files _added_ or _changed_, but since a file can influence an exercise, it's safer to run the actions for the _exercise_, if one of its files changes.
+
+It can be very helpful to make the actions that should run, available locally as well. This means that the scripts that do the actual work are also manually runnable. **Do not inline** the action inside the workflow files. For example, checking for stubs can be completely bashed out inside the workflow file, but the recommendation here is to create a new executable script `scripts/ci-check` instead.
+
+> "But the command is very short, e.g. `eslint . --ext ts --ext tsx`".
+>
+> When this command needs to be updated, it now needs to update in all the places in the documentation, the workflow files, ánd in the _minds of the maintainers_. Extracting this to a script resolves all that. Reading a workflow file can also be **very** daunting.
+
+### PR exercises changed
```
Comment (iHiD) [inline workflows/README.md:48]:
```suggestion
### Checks on PRs where exercises change
```

File: workflows/README.md [hunk 22]
```diff
@@ -0,0 +1,310 @@
...
+scripts/pr exercises/two-fer/README.md exercises/two-fer/.meta/example.ext
+```
+
+It's recommended to run any actions against the changed _exercise_ and not the changed _file_. This is because changing a file is likely to trigger changes for the entire exercise (think: configuration, packages).
+
+> **Not ready?** / **Complex?**
+>
+> Before implementing this optimization, it may be ignored! The migration guide hints at adding at a later stage. If the input arguments are ignored, all the checks are running on all the exercises. **This is perfectly fine**. It will just take longer.
+
+### Integrity
```
Comment (iHiD) [inline workflows/README.md:62]:
```suggestion
### Integrity checks
```

File: workflows/README.md [hunk 23]
```diff
@@ -0,0 +1,310 @@
...
+If the track has a single "top-level" dependency file and/or other configuration files, add an [integrity][wiki-integrity] step (to exist alongside a `scripts/sync` or `bin/sync`, which would copy all configuration files to all exercises) that ensures the top-level/base files are the same as the one copied to the exercise directories. Now dependencies can be updates, synced across the repository, and ensures that all exercises have the same configuration.
+
+A common way to accomplish this is to use a checksum. Ubuntu (and various other Linux distributions) comes with a tool called `sha1sum`, but using _whichever_ method to hash or reduce the configuration file (md5, sha1, crc32) to a checksum value, would work:
+
+```bash
+$ sha1sum README.md
+cd58091c5043bf21f00d39ff1740d8b2976deeff *README.md
+```
+
+### Security
```
Comment (iHiD) [inline workflows/README.md:73]:
```suggestion
### Security checks
```

File: workflows/README.md [hunk 24]
```diff
@@ -0,0 +1,310 @@
...
+
+It's recommended to run any actions against the changed _exercise_ and not the changed _file_. This is because changing a file is likely to trigger changes for the entire exercise (think: configuration, packages).
+
+> **Not ready?** / **Complex?**
+>
+> Before implementing this optimization, it may be ignored! The migration guide hints at adding at a later stage. If the input arguments are ignored, all the checks are running on all the exercises. **This is perfectly fine**. It will just take longer.
+
+### Integrity
+
+If the track has a single "top-level" dependency file and/or other configuration files, add an [integrity][wiki-integrity] step (to exist alongside a `scripts/sync` or `bin/sync`, which would copy all configuration files to all exercises) that ensures the top-level/base files are the same as the one copied to the exercise directories. Now dependencies can be updates, synced across the repository, and ensures that all exercises have the same configuration.
```
Comment (iHiD) [inline workflows/README.md:64]:
```suggestion
If the track has a single "top-level" dependency file and/or other configuration files, add an [integrity][wiki-integrity] step (that exists alongside a `scripts/sync` or `bin/sync`, which would copy all configuration files to all exercises), which ensures that the top-level/base files are the same as the one copied to the exercise directories. Now dependencies can be updated, synced across the repository, and we can ensure that all exercises have the same configuration.
```

File: workflows/README.md [hunk 25]
```diff
@@ -0,0 +1,310 @@
...
+A common way to accomplish this is to use a checksum. Ubuntu (and various other Linux distributions) comes with a tool called `sha1sum`, but using _whichever_ method to hash or reduce the configuration file (md5, sha1, crc32) to a checksum value, would work:
+
+```bash
+$ sha1sum README.md
+cd58091c5043bf21f00d39ff1740d8b2976deeff *README.md
+```
+
+### Security
+
+If the track uses additional workflows that require access to the GitHub token or other secrets, it's best practice to pin **all** actions used in the workflow to a specific commit. See [GitHub's security hardening guide][github-actions-security] for details.
```
Comment (iHiD) [inline workflows/README.md:75]: (FYI: https://github.com/exercism/problem-specifications/issues/1723)

File: workflows/README.md [hunk 26]
```diff
@@ -0,0 +1,310 @@
...
+
+For example:
+
+```diff
+- uses: julia-actions/setup-julia@v1
++ uses: julia-actions/setup-julia@d26d1111976eae5f00db04f0515ab744ec9cd79e # 1.3.1
+```
+
+If the tooling has lockfiles for dependency management, consider checking it into the repository and use a "frozen lockfile" inside the workflow files. For example: `npm ci`, `yarn install --frozen-lockfile` and `bundle install --frozen`. This ensures that the lockfile is up-to-date when changing dependencies and prevents malicious packages to come in.
```
Comment (iHiD) [inline workflows/README.md:84]: I'm not sure we care, as the point is clear, but Bundlers `--frozen` flag is deprecated.

File: workflows/README.md [hunk 27]
```diff
@@ -0,0 +1,310 @@
...
+
+## Templates
+
+In this directory at minimum there are the following templates:
+
+- `configlet.yml`: This workflow will do a fetch the latest configlet binary and lint this repository. Run on every commit. For PRs, runs on the actual commit and a "after merge" tree.
+- `ci.yml`: This workflow **only runs on `master`**, once on each commit.
```
Comment (iHiD) [inline workflows/README.md:91]:
```suggestion
- `ci.yml`: This workflow **only runs on the main branch (`master` or `main` by default)**, once on each commit.
```

**Review by iHiD — APPROVED** [review]
Loads of little comments - all for clarity, flow and grammar. The content is really fantastic. Thank you so much 💙

File: workflows/README.md (hunk 27 above)
Comment (SaschaMann) [reply workflows/README.md:91]: If you want to future-proof the workflows, you can replace `branches: [master]` with `branches: [master, main]`

File: workflows/README.md (hunk 19 above)
Comment (SleeplessByte) [reply workflows/README.md:18]: This should be `maintainers`. Track is very ambigious.

Maintainers _should_ be bound to code *style*, students *should not*.

File: workflows/README.md (hunk 26 above)
Comment (SleeplessByte) [reply workflows/README.md:84]: What's the "new" way?

File: workflows/README.md (hunk 27 above)
Comment (SleeplessByte) [reply workflows/README.md:91]: Yeah good idea. Lets' do that.

File: workflows/README.md [hunk 28]
```diff
@@ -0,0 +1,310 @@
...
+- # - scripts/ci
++ # - bin/ci-check
++ # - bin/ci
+
+- name: <track> / master
++ name: elm / master
+
+  on:
+    push:
+      branches: [master]
```
Comment (SleeplessByte) [inline workflows/README.md:200]:
```suggestion
      branches: [master, main]
```

File: workflows/ci.yml [hunk 29]
```diff
@@ -0,0 +1,64 @@
...
+# Requires scripts:
+# - scripts/ci-check
+# - scripts/ci
+
+name: <track> / master
+
+on:
+  push:
+    branches: [master]
```
Comment (SleeplessByte) [inline workflows/ci.yml:22]:
```suggestion
    branches: [master, main]
```

File: workflows/ci.yml [hunk 30]
```diff
@@ -0,0 +1,64 @@
...
+# Find Github Actions to setup tooling here:
+# - https://github.com/actions/?q=setup&type=&language=
+# - https://github.com/actions/starter-workflows/tree/main/ci
+# - https://github.com/marketplace?type=actions&query=setup
+#
+# Requires scripts:
+# - scripts/ci-check
+# - scripts/ci
+
+name: <track> / master
```
Comment (SleeplessByte) [inline workflows/ci.yml:18]:
```suggestion
name: <track> / main
```

"main" here is doubly the better default branch name, and means "default".

File: workflows/ci.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (SleeplessByte) [reply workflows/ci.yml:44]: Coming in a future PR 🏃

File: workflows/README.md (hunk 19 above)
Comment (SleeplessByte) [reply workflows/README.md:26]: (If you want the wording to be better, PRs appreciated ❤️ 🙉)

**Comment by SleeplessByte** [comment]
Thank you everyone who read along, gave suggestions, or/and pinged me on Slack.

Thanks @SaschaMann for your early help and thanks @iHiD for fixing the 5:00 English.

Y'all can now PR changes

File: workflows/README.md (hunk 26 above)
Comment (iHiD) [reply workflows/README.md:94]: `bundler config set x y` (e.g. https://github.com/exercism/v3-website/blob/master/docker/rails.Dockerfile#L16)

Basically, set it globally, rather than just on the one command.
